packages/@aws-cdk/aws-ecs/lib/base/task-definition.ts

```diff
@@ -293,6 +294,40 @@ export class TaskDefinition extends TaskDefinitionBase {
     return this._executionRole;
   }
 
+  /**
+   * Validate the existence of the input target and set default values.
+   */
+  public validateTarget(options: LoadBalancerTargetOptions): LoadBalancerTarget {
+    const targetContainer = this.findContainer(options.containerName);
+    if (targetContainer === undefined) {
+      throw new Error(`No container named '${options.containerName}'. Did you call "addContainer()"?`);
+    }
+    const targetProtocol = options.protocol || Protocol.TCP;
+    const targetContainerPort = options.containerPort || targetContainer.containerPort;
+    const portMapping = targetContainer._findPortMapping(targetContainerPort, targetProtocol);
+    if (portMapping === undefined) {
+      // tslint:disable-next-line:max-line-length
+      throw new Error(`Container '${targetContainer}' has no mapping for port ${options.containerPort} and protocol ${targetProtocol}. Did you call "container.addPortMapping()"?`);
+    }
+    return {
+      containerName: options.containerName,
+      portMapping
+    };
+  }
+
+  /**
+   * Returns the port range to be opened that match the provided container name and container port.
+   */
+  public findPortRange(portMapping: PortMapping): ec2.Port {
```

Comment:
It's not really "find"ing anything so I would call it something else. `portRangeFromPortMapping` or something?

Also please hide this from consumers.